### DISCUSSION TITLE
Value storage problem

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#23301] [Q&A Modules: Solid mechanics] [answered]
Hello All

I tried to print out equivalent_slip_increment in 'CrystalPlasticityStressUpdateBase.C'. I add some lines of codes as below:
`

    printf("equivalent_slip_increment(1,1) = %f\n",equivalent_slip_increment(0,0));
    printf("equivalent_slip_increment(1,2) = %f\n",equivalent_slip_increment(0,1));
    printf("equivalent_slip_increment(1,3) = %f\n",equivalent_slip_increment(0,2));
    printf("equivalent_slip_increment(2,1) = %f\n",equivalent_slip_increment(1,0));
    printf("equivalent_slip_increment(2,2) = %f\n",equivalent_slip_increment(1,1));
    printf("equivalent_slip_increment(2,3) = %f\n",equivalent_slip_increment(1,2));
    printf("equivalent_slip_increment(3,1) = %f\n",equivalent_slip_increment(2,0));
    printf("equivalent_slip_increment(3,2) = %f\n",equivalent_slip_increment(2,1));
    printf("equivalent_slip_increment(3,3) = %f\n",equivalent_slip_increment(2,2));
    printf("equivalent_slip_increment(1,4) = %f\n",equivalent_slip_increment(0,3));
    printf("equivalent_slip_increment(4,1) = %f\n",equivalent_slip_increment(3,0));

`
equivalent_slip_increment is a 3x3 matrix. But when I print out equivalent_slip_increment(1,4), it still has value.
![Screenshot from 2023-01-31 16-21-07](https://user-images.githubusercontent.com/81499506/215906262-341ce9ee-e36b-430d-a077-961afbbfc5f6.png)
Why it's out of dimension, but still has value?

Thanks
Pengfei

**Comment by GiudGiud** [comment, marked as answer]
If you are truly out of bounds, you could be printing part of the next tensor.
based on the other thread, you are using a vector of tensor material property

In general you should not print out of bounds

**Reply by pshen20127** [reply]
Get it, thank you.
